core/src/main/java/org/apache/iceberg/hadoop/Util.java

```diff
@@ -36,4 +47,21 @@ public static FileSystem getFs(Path path, Configuration conf) {
       throw new RuntimeIOException(e, "Failed to get file system for path: %s", path);
     }
   }
+
+  public static String[] blockLocations(CombinedScanTask task, Configuration conf) {
+    Set<String> locationSets = Sets.newHashSet();
+    for (FileScanTask f : task.files()) {
+      Path path = new Path(f.file().path().toString());
+      try {
+        FileSystem fs = path.getFileSystem(conf);
+        for (BlockLocation b : fs.getFileBlockLocations(path, f.start(), f.length())) {
+          locationSets.addAll(Arrays.asList(b.getHosts()));
+        }
+      } catch (IOException ioe) {
+        LOG.warn("Failed to get block locations for path {}", path, ioe);
```

Comment:
Is it OK if this happens? We don't want to rethrow this?

Reply:
For now, I've kept it as it was for Spark. Seem like locality is not really needed for cloud stores and it is an optimization for HDFS. I can throw an exception and then handle it across Spark and MR if you guys think this is necessary..

Reply:
Locality preference is just a nice-to-have thing, to throw an exception for this seems not so necessary.

Reply:
OK, makes sense.

Reply:
Yes, locality is optional and not being able to get it should not cause a job to fail.